### PR TITLE
Telekinesis can throw non items (vending machines, lockers) only 1 tile far

### PR DESCRIPTION
**Pull request by Gamer025**
## About The Pull Request
Telekinesis users can now only throw items 10 tiles, everything else can only be thrown 1 tile

## Why It's Good For The Game
This balances the fact that you can throw vending machines at people and crush them
It also is a bit more realistic that you can only throw heavy stuff like lockers by 1 tile

## Changelog
:cl: Gamer025
tweak: Telekinesis users can now only throw items 10 tiles far, everything else (like lockers, machines, etc) can only be thrown 1 tile far
/:cl:

### CODE REVIEW

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -1011,6 +1011,10 @@ GLOBAL_LIST_EMPTY(vending_products)
 		tilt(fatty=hit_atom)
 	return ..()
 
+/obj/machinery/vending/attack_tk_grab(mob/user)
+	to_chat(user, "<span class='warning'>The vending machine is too heavy.</span>")
```
Comment (Jared-Fogle) [inline code/modules/vending/_vending.dm:1015]:
```suggestion
	to_chat(user, "<span class='warning'>[src] is too heavy.</span>")
```

**Review by Mothblocks — DISMISSED** [review]

**Comment by cacogen** [comment]
It'd be better if it was nerfed than removed

**Comment by Jared-Fogle** [comment]
pull requests are always open

**Comment by cacogen** [comment]
I know but

**Comment by AlinaStarkova** [comment]
personally, this is just a tweak, not a fix, i dont think being able to throw vending machines with your brain power is a bug at all, OP maybe, but not a bug, unless the "fix" is being able to throw them if they are secured to the ground

also not a tweak i like, OP it may be, a lot of things are OP in the game, and they add fun, i dont think the rare few times someone actually bothers to do this warrants a nerf, a tweak for this i would agree with however, is only being able to throw it just 1 tile at a time like all heavy objects, though im assuming this is current function?

**Comment by Jared-Fogle** [comment]
>  i dont think being able to throw vending machines with your brain power is a bug at all,

The PR that did this was requested by an administrator for administrative purposes, and the OP explicitly said in the replies he didn't think it was possible. Yes, it's a bug.

**Comment by Gamer025** [comment]
> personally, this is just a tweak, not a fix, i dont think being able to throw vending machines with your brain power is a bug at all, OP maybe, but not a bug, unless the "fix" is being able to throw them if they are secured to the ground
> 
> also not a tweak i like, OP it may be, a lot of things are OP in the game, and they add fun, i dont think the rare few times someone actually bothers to do this warrants a nerf, a tweak for this i would agree with however, is only being able to throw it just 1 tile at a time like all heavy objects, though im assuming this is current function?

I actually like this idea and I think together with the fact that you still need to manually unwrench the vending machine before being able to throw it I think this is fine balance wise.
So I changed it so that TK can only throw /obj/items 10 tiles and everything else can only be thrown 1 tile (aka stuff like lockers and vending machines)

**Comment by AlinaStarkova** [comment]
> I actually like this idea and I think together with the fact that you still need to manually unwrench the vending machine before being able to throw it I think this is fine balance wise.
> So I changed it so that TK can only throw /obj/items 10 tiles and everything else can only be thrown 1 tile (aka stuff like lockers and vending machines)

if you do it like that, make sure the big things have a cooldown, and cant just be spammed to move it fast, though i think TK has a cooldown built into it

**Comment by Gamer025** [comment]
> > I actually like this idea and I think together with the fact that you still need to manually unwrench the vending machine before being able to throw it I think this is fine balance wise.
> > So I changed it so that TK can only throw /obj/items 10 tiles and everything else can only be thrown 1 tile (aka stuff like lockers and vending machines)
> 
> if you do it like that, make sure the big things have a cooldown, and cant just be spammed to move it fast, though i think TK has a cooldown built into it

Yeah I tested it and there was a cooldown, you can kinda throw the vending machine as fast they can move in the new brand intelligence event

File: code/_onclick/telekinesis.dm [hunk 2]
```diff
@@ -171,7 +172,11 @@
 	else
 		. = TRUE
 		apply_focus_overlay()
-		focus.throw_at(target, 10, 1,user)
+		//Only items can be thrown 10 tiles everything else only 1 tile
+		if (isitem(focus))
+			focus.throw_at(target, 10, 1,user)
+		else
+			focus.throw_at(target, 1, 1,user)
```
Comment (Jared-Fogle) [inline code/_onclick/telekinesis.dm:179]: I'd rather this be a var on /atom/movable.

Comment (Gamer025) [reply code/_onclick/telekinesis.dm]: That is properly the cleanest way of doing this, created tk_throw_range for this

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**